Sources/Articles/Documentation.docc/2024/beginners-guide-to-protocol-buffers-and-grpc-with-swift/beginners-guide-to-protocol-buffers-and-grpc-with-swift.md

```diff
@@ -0,0 +1,382 @@
+        .package(url: "https://github.com/grpc/grpc-swift.git", from: "1.23.0"),
+    ],
+    targets: [
+        .executableTarget(
+            name: "App",
+            dependencies: [
+                .product(name: "ArgumentParser", package: "swift-argument-parser"),
+
+                .product(name: "GRPC", package: "grpc-swift"),
+                .product(name: "_GRPCCore", package: "grpc-swift"),
+            ],
+            plugins: [
+                .plugin(name: "SwiftProtobufPlugin", package: "swift-protobuf"),
+                .plugin(name: "GRPCSwiftPlugin", package: "grpc-swift"),
+            ]
+        ),
+    ]
+)
+```
+
+The next step is to add the `todo_messages.proto` file into the `Sources/App` directory. Additionally, create a `swift-protobuf-config.json` file to configure the Swift Protobuf plugin:
+
+```json
+{
+    "invocations": [
+        {
+            "protoFiles": [
+                "todo_messages.proto",
+            ],
+            "visibility": "internal",
+            "server": false,
+        },
+    ]
+}
+```
+
+This configuration specifies how the Swift Protobuf plugin will generate code. The `protoFiles` array includes the `todo_messages.proto` file, which will be used as the input for code generation. The `visibility` setting is defined as `internal`, and the `server` flag is set to `false`, indicating that the plugin will generate only the data models without generating any server-side code.
+
+A similar configuration is required for the gRPC Swift plugin. Add the `todo.proto` file to the source directory, and create a `grpc-swift-config.json` file with the following contents:
+
+```json
+{
+    "invocations": [
+        {
+            "protoFiles": [
+                "todo.proto",
+            ],
+            "visibility": "internal",
+            "server": true,
+            "client": false,
+            "keepMethodCasing": false,
+            "reflectionData": true
+        }
+    ]
+}
+```
+
+By using the Swift gRPC plugin, server-side Swift code is generated, and it utilizes the data types produced by the Swift Protobuf plugin.
+
+It's important to note that the Swift Protobuf plugin generates only the data structures. In contrast, the Swift gRPC plugin can generate both the server and client-side interfaces for the whole communication layer. Simply put, Protocol Buffers manage the data encoding and decoding, while gRPC uses Protocol Buffers to enable RPC communication.
+
+One caveat when using Xcode is that the plugins might need to be manually enabled. This can be done in the Report Navigator by clicking the links below the log messages. Another [issue](https://forums.swift.org/t/plugin-doesnt-have-access-to-binary-package-manager-extensible-build-tools-se-0303-and-se-0305/56038/10) is that Xcode may not access external build tools, so linking the protoc command manually may be required using the following command:
```

Comment:
```suggestion
One caveat when using Xcode is that the plugins need to be manually enabled. This can be done in the Report Navigator by clicking the links below the log messages. Another [issue](https://forums.swift.org/t/plugin-doesnt-have-access-to-binary-package-manager-extensible-build-tools-se-0303-and-se-0305/56038/10) is that Xcode may not access external build tools, so linking the protoc command manually may be required using the following command:
```